### PR TITLE
Introduce StructOpt and refactor cargo pants

### PR DESCRIPTION
**Pull request by amy-keibler**
Replicate the existing CLI experience from clap, but without the stringly-typed
interface.

This pull request makes the following changes:
* Switches cargo-pants to use StructOpt and re-structure its `bin/` folder layout
* StructOpt enables env fallback, so I added the option `--ossi-api-key` for someone to manually pass it in. This will fall-back to `OSSI_API_KEY`, which removes the need for the explicit env query.

There should be no significant changes in behavior besides a slightly different `--help` output when running `cargo pants --help` and running the raw program now outputs information about the `pants` subcommand instead of a warning message.

### CODE REVIEW

File: src/bin/iq/main.rs [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+                    iq_bar.set_style(spinner_style.clone());
+                    iq_bar.set_message(format!(
+                        "{}{}",
+                        SPARKIES, "Sending SBOM to Nexus IQ Server for evaluation"
+                    ));
+                    let iq =
+                        IQClient::new(server_url, username, token, stage, application, attempts);
+                    match iq.audit_with_iq_server(sbom) {
+                        Ok(res) => {
+                            trace!("Response recieved: {:#?}", res.url_results);
```
Comment (bhamail) [inline src/bin/iq/main.rs]: minor: typo

```suggestion
                            trace!("Response received: {:#?}", res.url_results);
```

File: src/bin/iq/main.rs [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+                            if res.url_results.is_error {
+                                panic!("{}", res.url_results.error_message.unwrap());
+                            }
+
+                            iq_bar.finish_with_message(format!(
+                                "{}{}",
+                                CRAB, "Nexus IQ Results obtained"
+                            ));
+                            println!("");
```
Comment (bhamail) [inline src/bin/iq/main.rs]: minor: remove ""

```suggestion
                            println!();
```

File: src/bin/iq/main.rs [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+                            process::exit(1);
+                        }
+                    }
+                }
+                Err(e) => {
+                    package_bar.finish_with_message(format!(
+                        "{}{}",
+                        CROSS_MARK, "Unable to obtain package list"
+                    ));
+                    println!("{}", e);
```
Comment (bhamail) [inline src/bin/iq/main.rs:186]: IDEA is flagging this, but it I'm not sure it makes sense to me.

     `Error` doesn't implement `Display` (required by {})

The higher `Err(e)` shows a type of `Err(e) Box<dyn Error>`, so maybe that is why the higher one (line 168) is not getting flagged? 
Just asking for my own knowledge - not a blocker seems to me. Dunno if we exercise this code.

File: src/bin/iq/main.rs [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        .into_iter()
+        .filter(|p| p.violations.as_ref().map_or(false, |v| !v.is_empty()))
+        .collect();
+
+    if policy_violations.len() > 0 {
+        println!(
+            "Components ({}) with policy violations found",
+            policy_violations.len()
+        );
+        println!("");
```
Comment (bhamail) [inline src/bin/iq/main.rs]: minor: delete ""

```suggestion
        println!();
```

File: src/bin/iq/main.rs [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+                        "Known violations: {}",
+                        violations
+                            .into_iter()
+                            .map(|v| policy_violation_to_styled_object(v.policy_name).to_string())
+                            .collect::<Vec<String>>()
+                            .join(",")
+                    );
+                    println!("Inverse Dependency graph");
+                    assert!(parser.print_the_graph(comp.package_url).is_ok());
+                    println!("");
```
Comment (bhamail) [inline src/bin/iq/main.rs]:
```suggestion
                    println!();
```

File: src/bin/iq/main.rs [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+
+fn print_iq_summary(
+    emoji: Emoji,
+    summary_line: StyledObject<&str>,
+    policy_violations: OpenPolicyViolations,
+    server: String,
+    html_url: String,
+) {
+    println!("{}{}", emoji, summary_line);
+    println!("");
```
Comment (bhamail) [inline src/bin/iq/main.rs]:
```suggestion
    println!();
```

File: src/bin/iq/main.rs [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+    generate_summary_table(policy_violations);
+    println!("");
```
Comment (bhamail) [inline src/bin/iq/main.rs]:
```suggestion
    println!();
```

File: src/bin/pants/main.rs [hunk 8]
```diff
@@ -11,97 +11,62 @@
...
+fn audit(
+    toml_file_path: String,
+    oss_index_api_key: Option<String>,
+    verbose_output: bool,
+    enable_color: bool,
+    include_dev: bool,
+) -> ! {
     let mut parser = ParseCargoToml::new(toml_file_path.clone(), include_dev);
     let packages = match parser.get_packages() {
         Ok(packages) => packages,
```
Comment (bhamail) [inline src/bin/pants/main.rs:72]: also please ignore similar to prior `Err(e)` question:

IDEA is flagging this (line 74 - not shown in diff), but it I'm not sure it makes sense to me.

     `Error` doesn't implement `Display` (required by {})

Just asking for my own knowledge - not a blocker seems to me. Dunno if we exercise this code.

**Review by bhamail — APPROVED** [review]
Very nice. Great to see so much code disappear!

Just some silly comments, feel free to ignore and/or educate me as you see fit.

(Minor tweaks to prints would normally be left out of the PR for clarity. I just flagged 'em because I noticed them. Ignore as desired.)

File: src/bin/iq/main.rs (hunk 3 above)
Comment (amy-keibler) [reply src/bin/iq/main.rs:186]: This was code that was moved and git failed to track it, so it has been working previously. I'm going to create an issue to follow up with this specifically in a separate branch

**Pull request merged**